### PR TITLE
tests(insert): add insert tests for json, text array and integer array

### PR DESCRIPTION
**Pull request by minh-hoang-trinh**
TLDR: add various tests case for the PR https://github.com/knex/knex/pull/5431 (rollback of https://github.com/knex/knex/pull/5321)

- insert tests for array of object into `text array` column #5430
- insert tests for array of integer into `integer array` column #5365
- insert tests for array of string into `text array` column #5365
- insert tests for json object into `jsonb` column

the PR https://github.com/knex/knex/pull/5431 by @caseywebdev should be merged first

A little bit more explanation: 

#5365 : user want to insert array of string into `text array` column, or array of integer into `integer array` column, which is valid and should be inserted, but the PR https://github.com/knex/knex/pull/5321 merged sometime ago break it (it will always stringify the array to json string)

@kibertoad At first I thought I can add a check for Array containing at least one plainObject with my previous PR https://github.com/knex/knex/pull/5444, but turn out it will not solve the #5430, my previous changes will need user to input a valid array of string into `text array` column

#5430 : user want to insert array of object into `text array` column (worked in previous version of knex). the PR https://github.com/knex/knex/pull/5321 break it (it will always stringify the array of object to json string)

### CODE REVIEW

**Comment by kibertoad** [comment]
@minh-hoang-trinh Fix was merged with https://github.com/knex/knex/pull/5439, but additional tests are very welcome. Could you please resolve conflicts in your PR, then I could merge it?

**Comment by minh-hoang-trinh** [comment]
> @minh-hoang-trinh Fix was merged with #5439, but additional tests are very welcome. Could you please resolve conflicts in your PR, then I could merge it?

thank @kibertoad, conflicts resolved ^^

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Thank you!
